CMakeLists.txt

```diff
@@ -4,24 +4,40 @@ project(nps_uw_sensors_gazebo)
 find_package(catkin REQUIRED tf)
 
 find_package(gazebo REQUIRED)
+find_package(roscpp REQUIRED)
+find_package(std_msgs REQUIRED)
 find_package(OpenCV)
+include_directories(${roscpp_INCLUDE_DIRS})
+include_directories(${std_msgs_INCLUDE_DIRS})
 include_directories(
   include
   ${catkin_INCLUDE_DIRS} 
   ${GAZEBO_INCLUDE_DIRS}
 )
 link_directories(${GAZEBO_LIBRARY_DIRS})
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${GAZEBO_CXX_FLAGS}")
 
 catkin_package()
 
 ## Plugins
-add_library(nps_gazebo_ros_gpu_sonar_single_beam_plugin
-            src/nps_gazebo_ros_gpu_sonar_single_beam.cpp)
-target_link_libraries(nps_gazebo_ros_gpu_sonar_single_beam_plugin
-                      GpuRayPlugin ${catkin_LIBRARIES})
+add_library(gazebo_ros_pulse_lidar_plugin src/gazebo_ros_pulse_lidar.cpp)
+add_library(nps_gazebo_ros_gpu_sonar_single_beam_plugin src/nps_gazebo_ros_gpu_sonar_single_beam.cpp)
+target_link_libraries(gazebo_ros_pulse_lidar_plugin nps_gazebo_ros_gpu_sonar_single_beam_plugin
+                      GpuRayPlugin ${catkin_LIBRARIES} ${GAZEBO_LIBRARIES} ${roscpp_LIBRARIES})
 
 # for launch
 install(DIRECTORY launch
         DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
         PATTERN "*~" EXCLUDE)
 
+# Build the stand-alone test program
+add_executable(lidar_test src/lidar_test.cc)
+
+if (${gazebo_VERSION_MAJOR} LESS 6)
+  # These two
```

Comment:
Lets not add any new code that supports Gazebo version less than 9.  I don't mind leaving alone existing C++ ifdefs.  Wiki indicates we support Gazebo 9+ for ROS Melodic.

Reply:
Resolved.